### ISSUE TITLE
Bullet physics stacking and multiple convex collsionshape problem and godotphysics stacking lag 

### ISSUE THREAD

**Issue by KirinMakes** [godotengine/godot#50316]
### Godot version

3.3.2.stable.official

### System information

Windows 10, GLES3, AMD RX 5700 XT

### Issue description

Bullet physics: Problem with object stacking (space between objects) of donut model which has collision shapes as multiple convex polygons (4 in this case). The shapes does not fit the model perfectly but the stick is small enough to go through hole of donut.

Godotphysics: Stacking works fine and donut hole goes through stick. No stacking problem. Stacking is laggy in godotphysics

Expected stacking to work and multiple convex collision shapes to work. Also stacking reching steady state is laggy

Video demonstrates issue. See also minimal reproduction project attached below.

[Video Demonstration](https://youtu.be/1FBVBSCZdy0)

### Steps to reproduce

Run using godotphysics and bullet physics to see each issue

Steps:

- Model is exported from blender as glb
- Imported to godot and made root node rigidbody (to work with physics). A reimport occured
- Create a new scene with the model and used mesh command on meshinstance to create multiple convex collision shapes as siblings
- used scene in main scene. 
- Change to bullet physics and godotphysics to see each problem.

### Minimal reproduction project

[PhysicsIssue.zip](https://github.com/godotengine/godot/files/6792103/PhysicsIssue.zip)

**Comment by Calinou** [comment]
cc @pouleyKetchoupp

**Comment by pouleyKetchoupp** [comment]
**For the collision issue in Bullet:**
It happens because the donut's size is very low. Bullet uses an extra margin for all collisions for optimization purpose, and it's causing collision to happen when it shouldn't.
Godot physics doesn't have this issue, so it's easier to setup and more flexible, but not as good for performance.
You can work around it by setting your whole scene at a bigger scale. One way to do that is to just change the scale at the root Spatial of your scene to (10, 10, 10), and adjust the gravity scale on your donuts to avoid slow motion, it seems to solve the issue.
Here's a modified version of your project as an example: [DonutPhysicsIssue_Scaled.zip](https://github.com/godotengine/godot/files/6794467/DonutPhysicsIssue_Scaled.zip)

**For the lag in Godot Physics:**
The poor performance is because the generated convex collision shapes are a bit too complex (4 pieces with 64 vertices each).
In the future, we'll add options to the convex shape generation to control the result a bit better, which might help a little bit.
In this case though, I would generally recommend making proper collision shapes by hand instead of using the automatic convex generation when you need accurate collision for objects used for gameplay. Even in Bullet you might start having performance issues with more objects involved, and the collision doesn't seem very accurate.
For example, you can make something like that using capsules and the generated convex shapes for comparison:

| Visual  | Collision |
| ------------- | ------------- | 
| <img width="197" alt="donut_visual" src="https://user-images.githubusercontent.com/1075032/125148391-c65cc300-e0e6-11eb-870f-c7b7d02ffeda.PNG">  | <img width="190" alt="donut_collision" src="https://user-images.githubusercontent.com/1075032/125148392-c957b380-e0e6-11eb-87c0-4e918d0b84ed.PNG"> |


Here's the modified project with capsules: [DonutPhysicsIssue_Capsules.zip](https://github.com/godotengine/godot/files/6794493/DonutPhysicsIssue_Capsules.zip)

Let us know if that helps!

**Comment by KirinMakes** [comment]
The projects attached does solve the individual physics issues. I found that if the root for the scene is scaled to 10 as in the first attached problem, godotphysics does not scale the donuts to be 10 times larger (even though the editor shows that it is 10 times larger), which is a little bit weird. For bullet physics it does the scaling. 

I checked the transform in the inspector for godotphysics for all rigidbodies and they are set to 1. It does work fine with the scale set to 1 though.

Other than that it seems godotphysics misses collison on the stick for the second-above-the-stick-donut when the gravity scale is set to default value of 1. I guess it gets too high of a velocity for the discrete collsion detection of godotphysics to handle.

Thanks for the help.
